content/6_Advanced/String_Search.mdx

```diff
@@ -44,8 +51,61 @@ string.
 	 />
 </Resources>
 
+Define an array $\pi_S$ of size $|S|$ such that $\pi_S[i]$ is equal to the length of the longest nontrivial suffix of the prefix ending at position $i$ the coincides with a prefix of the entire string. Formally,
```

Comment:
dustin, these explanation is crap
either fix it so normal people can read it or add some examples

Reply:
look its not my job to explain the z algorithm
thats the purpose of the module
i might add the detailed description later

Reply:
> look its not my job to explain the z algorithm

then what's this part for????

Reply:
wait srry wrong thing

Reply:
the formally parts clarifies it?

Reply:
dustin i don't know about you but half of my braincells die whenever i see an equation longer than 5 or 6 characters

Reply:
that seems like a you problem

Reply:
dustin not everyone's a math genius just add some examples or a better explanation, it isn't that hard
don't make me pull out my mod powers

Reply:
added more examples, should be more clear now

Reply:
> dustin i don't know about you but half of my braincells die whenever i see an equation longer than 5 or 6 characters

this is a you problem and not a dustin problem
but dustin make the equations easier to understand (this is CP not Math)